.github/workflows/build.yaml

```diff
@@ -0,0 +1,66 @@
+# Copyright 2024 NVIDIA CORPORATION
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Run this workflow on pull requests or merge to main/release branches
+name: build
+
+on:
+  pull_request:
+    types:
+      - opened
+      - synchronize
+    branches:
+      - main
+      - release-*
+
+jobs:
```

Comment:
See https://github.com/NVIDIA/libnvidia-container/pull/239 and https://github.com/NVIDIA/nvidia-container-toolkit/pull/332 for a way to selectively run entries in the parallel matrix on PRs. I would rather do that here too than maintain two separate workflows.

Reply:
I don't see why not, my initial intention was to directly translate the gilab pipeline, and optimize it later down the road. but I like the approach